subprojects/docs/src/docs/release/notes.md

```diff
@@ -14,6 +14,14 @@ Add-->
 
 Starting with this release, it is now possible to use SNAPSHOT plugin versions in the `plugins {}` and `pluginManagement {}` blocks.
 
+### Periodic cache cleanup
+
+Caching has always been one of the strong suits of Gradle. Over time, more and more persistent caches have been added to improve performance and support new features, requiring more and more disk space on build servers and developer workstations. Gradle now addresses one of the most highly voted issues on GitHub and introduces the following cleanup strategies:
+
+- Version-specific cache directories in `GRADLE_USER_HOME/caches/<gradle-version>/` are checked periodically (at most every 24 hours) for whether they are still in use. If not, directories for release versions are deleted after 30 days of inactivity, snapshot versions after 7 days of inactivity. Moreover, the corresponding Gradle distributions at `GRADLE_USER_HOME/wrapper/dists/gradle-<gradle-version>-{bin,all}/` are deleted as well, if present.
```

Comment:
Reading this makes me wonder if we are covered when users have their own custom Gradle distribution.

Reply:
Right, if they use a different naming pattern, we're not going to find and delete the distribution.

Reply:
This PR adds support for custom distributions: #5906.